### ISSUE TITLE
Failed to load user settings

### ISSUE THREAD

**Issue by nataziel** [devcontainers/cli#783]
I installed the CLI via the VSCode command, `>Dev Containers: Install devcontainer CLI`

Everything seems to run fine, I'm able to build containers, connect to them etc.

However, every time I run a devcontainer command some errors are thrown:

```bash
devcontainer --version
Failed to load user settings
Error: Failed to determine user settings path from: C:/Users/me/AppData/Roaming/Code/User/globalStorage/ms-vscode-remote.remote-containers/cli-bin/devcontainer
    at d7 (c:\Users\me\.vscode\extensions\ms-vscode-remote.remote-containers-0.348.0\dev-containers-user-cli\dist\node\devContainersUserCLI.js:228:1877)
    at kbe (c:\Users\me\.vscode\extensions\ms-vscode-remote.remote-containers-0.348.0\dev-containers-user-cli\dist\node\devContainersUserCLI.js:227:5182)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
0.56.2
```

```bash
devcontainer open
Failed to load user settings
Error: Failed to determine user settings path from: C:/Users/me/AppData/Roaming/Code/User/globalStorage/ms-vscode-remote.remote-containers/cli-bin/devcontainer
    at d7 (c:\Users\me\.vscode\extensions\ms-vscode-remote.remote-containers-0.348.0\dev-containers-user-cli\dist\node\devContainersUserCLI.js:228:1877)
    at a7 (c:\Users\me\.vscode\extensions\ms-vscode-remote.remote-containers-0.348.0\dev-containers-user-cli\dist\node\devContainersUserCLI.js:227:5309)
    at Object.handler (c:\Users\me\.vscode\extensions\ms-vscode-remote.remote-containers-0.348.0\dev-containers-user-cli\dist\node\devContainersUserCLI.js:227:4647)
    at c:\Users\me\.vscode\extensions\ms-vscode-remote.remote-containers-0.348.0\dev-containers-user-cli\dist\node\devContainersUserCLI.js:82:27312
    at Ks (c:\Users\me\.vscode\extensions\ms-vscode-remote.remote-containers-0.348.0\dev-containers-user-cli\dist\node\devContainersUserCLI.js:82:23370)
    at c_.handleValidationAndGetResult (c:\Users\me\.vscode\extensions\ms-vscode-remote.remote-containers-0.348.0\dev-containers-user-cli\dist\node\devContainersUserCLI.js:82:27295)
    at c_.applyMiddlewareAndGetResult (c:\Users\me\.vscode\extensions\ms-vscode-remote.remote-containers-0.348.0\dev-containers-user-cli\dist\node\devContainersUserCLI.js:82:27915)
    at c_.runCommand (c:\Users\me\.vscode\extensions\ms-vscode-remote.remote-containers-0.348.0\dev-containers-user-cli\dist\node\devContainersUserCLI.js:82:25658)
    at [runYargsParserAndExecuteCommands] (c:\Users\me\.vscode\extensions\ms-vscode-remote.remote-containers-0.348.0\dev-containers-user-cli\dist\node\devContainersUserCLI.js:140:7244)
    at b_.parse (c:\Users\me\.vscode\extensions\ms-vscode-remote.remote-containers-0.348.0\dev-containers-user-cli\dist\node\devContainersUserCLI.js:138:1731)
ApplicationInsights:Failed to read JSON config file:  [
  Error: ENOENT: no such file or directory, open 'c:\Users\me\.vscode\extensions\ms-vscode-remote.remote-containers-0.348.0\dev-containers-user-cli\applicationinsights.json'
      at Object.openSync (node:fs:605:3)
      at Object.func [as openSync] (node:electron/js2c/node_init:2:2107)
      at Object.readFileSync (node:fs:473:35)
      at Object.readFileSync (node:electron/js2c/node_init:2:9456)
      at t._loadJsonFile (c:\Users\me\.vscode\extensions\ms-vscode-remote.remote-containers-0.348.0\dev-containers-user-cli\dist\node\devContainersUserCLI.js:32:5306)
      at new t (c:\Users\me\.vscode\extensions\ms-vscode-remote.remote-containers-0.348.0\dev-containers-user-cli\dist\node\devContainersUserCLI.js:32:4941)
      at t.getInstance (c:\Users\me\.vscode\extensions\ms-vscode-remote.remote-containers-0.348.0\dev-containers-user-cli\dist\node\devContainersUserCLI.js:32:5022)
      at c:\Users\me\.vscode\extensions\ms-vscode-remote.remote-containers-0.348.0\dev-containers-user-cli\dist\node\devContainersUserCLI.js:33:95718
      at c:\Users\me\.vscode\extensions\ms-vscode-remote.remote-containers-0.348.0\dev-containers-user-cli\dist\node\devContainersUserCLI.js:5:244
      at c:\Users\me\.vscode\extensions\ms-vscode-remote.remote-containers-0.348.0\dev-containers-user-cli\dist\node\devContainersUserCLI.js:38:32266 {
    errno: -4058,
    syscall: 'open',
    code: 'ENOENT',
    path: 'c:\\Users\\me\\.vscode\\extensions\\ms-vscode-remote.remote-containers-0.348.0\\dev-containers-user-cli\\applicationinsights.json'
  }
]
ApplicationInsights:Invalid JSON config file:  [
  SyntaxError: Unexpected end of JSON input
      at JSON.parse (<anonymous>)
      at t._loadJsonFile (c:\Users\me\.vscode\extensions\ms-vscode-remote.remote-containers-0.348.0\dev-containers-user-cli\dist\node\devContainersUserCLI.js:32:5401)
      at new t (c:\Users\me\.vscode\extensions\ms-vscode-remote.remote-containers-0.348.0\dev-containers-user-cli\dist\node\devContainersUserCLI.js:32:4941)
      at t.getInstance (c:\Users\me\.vscode\extensions\ms-vscode-remote.remote-containers-0.348.0\dev-containers-user-cli\dist\node\devContainersUserCLI.js:32:5022)
      at c:\Users\me\.vscode\extensions\ms-vscode-remote.remote-containers-0.348.0\dev-containers-user-cli\dist\node\devContainersUserCLI.js:33:95718
      at c:\Users\me\.vscode\extensions\ms-vscode-remote.remote-containers-0.348.0\dev-containers-user-cli\dist\node\devContainersUserCLI.js:5:244
      at c:\Users\me\.vscode\extensions\ms-vscode-remote.remote-containers-0.348.0\dev-containers-user-cli\dist\node\devContainersUserCLI.js:38:32266
      at c:\Users\me\.vscode\extensions\ms-vscode-remote.remote-containers-0.348.0\dev-containers-user-cli\dist\node\devContainersUserCLI.js:5:244
      at c:\Users\me\.vscode\extensions\ms-vscode-remote.remote-containers-0.348.0\dev-containers-user-cli\dist\node\devContainersUserCLI.js:69:26198
      at c:\Users\me\.vscode\extensions\ms-vscode-remote.remote-containers-0.348.0\dev-containers-user-cli\dist\node\devContainersUserCLI.js:5:244
]
```

`c:\\Users\\me\\.vscode\\extensions\\ms-vscode-remote.remote-containers-0.348.0\\dev-containers-user-cli\\applicationinsights.json` doesn't exist

I'm not sure if this is related but when I run `devcontainer open` it works and opens VSCode connected to the dev container but it prompts me to rebuild the container saying that the specification has changed when I know that it hasn't.

Any tips?

**Comment by chrmarti** [comment]
What kind of shell are you running it from? The following shows slashes in the path where I would expect backslashes:
```
Error: Failed to determine user settings path from: C:/Users/me/AppData/Roaming/Code/User/globalStorage/ms-vscode-remote.remote-containers/cli-bin/devcontainer
```

Adding a fix for the `applicationinsights.json doesn't exist` error.

**Comment by nataziel** [comment]
I'm running it from Git-Bash (on windows)

**Comment by nataziel** [comment]
@chrmarti, the latest VSCode update has updated the CLI to 0.58.0, but I'm still getting this error:
```
Failed to load user settings
Error: Failed to determine user settings path from: C:/Users/me/AppData/Roaming/Code/User/globalStorage/ms-vscode-remote.remote-containers/cli-bin/devcontainer
    at E7 (c:\Users\me\.vscode\extensions\ms-vscode-remote.remote-containers-0.354.0\dev-containers-user-cli\dist\node\devContainersUserCLI.js:229:1877)
    at zbe (c:\Users\me\.vscode\extensions\ms-vscode-remote.remote-containers-0.354.0\dev-containers-user-cli\dist\node\devContainersUserCLI.js:228:5242)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
```

**Comment by chrmarti** [comment]
@nataziel Please try Dev Containers 0.355.0-pre-release. The fix didn't make it into 0.354.0.

**Comment by nataziel** [comment]
@chrmarti pre-release version works like a charm. Thanks!
